Production_scripts/B05b_Link_CUP_marker.R

```diff
@@ -0,0 +1,145 @@
+#####################################################
+# A&E Extract
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description - Process A & E extract
+#####################################################
+
+# Load packages
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+library(createslf)
+library(slfhelper)
+library(readr)
+
+# Read in data---------------------------------------
+
+# Specify year
+year <- 1920
+
+## get data ##
+ae_cup_extract <- readr::read_csv(
+  file =
+    paste0(
+      get_year_dir(year = latest_year),
+      "/Extracts/A&E-UCD-CUP-extract-20",
+      latest_year,
+      ".csv.gz"
+    )
+) %>%
+  # rename
+  rename(
+    record_keydate1 = "ED Arrival Date",
+    keyTime1 = "ED Arrival Time",
+    case_ref_number = "ED Case Reference Number [C]",
+    cup_marker = "CUP Marker",
+    cup_pathway = "CUP Pathway Name"
+  ) %>%
+  # date type
+  mutate(
+    record_keydate1 = as.Date(record_keydate1)
+  )
+
+
+## sort for linking onto data extract ##
+# remove any duplicates
+ae_cup_extract <-
+  ae_cup_extract %>%
+  arrange(record_keydate1, keyTime1, case_ref_number) %>%
+  group_by(record_keydate1, keyTime1, case_ref_number) %>%
+  mutate(
+    cup_marker = first(cup_marker),
+    cup_pathway = first(cup_pathway)
+  ) %>%
+  ungroup()
+
+
+## match files ##
+matched_data <-
+  outfile %>%
+  full_join(ae_cup_extract, by = c("record_keydate1", "keyTime1", "case_ref_number")) %>%
+  arrange(chi, record_keydate1, keyTime1, record_keydate2, keyTime2)
+
```

Comment:
Made a new section for joining the data. Read in the processed  A&E extract here